lib/utils/mpi_utils.cpp

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2023, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: Utility functions for MPI distribution.
+
+#include "mpi_utils.h"
+#include "Utilities.h"
+
+#include <iomanip>
+#include <stdlib.h>
+#include <sys/stat.h>
+
+namespace CAROM {
+
+int
+split_dimension(const int &dim, const MPI_Comm &comm)
+{
+    int mpi_init;
+    MPI_Initialized(&mpi_init);
+    CAROM_VERIFY(mpi_init != 0);
+
+    int d_num_procs, d_rank;
+    MPI_Comm_rank(comm, &d_rank);
+    MPI_Comm_size(comm, &d_num_procs);
+
+    int local_dim = dim / d_num_procs;
+    if (dim % d_num_procs > d_rank)
+        local_dim++;
+
+    return local_dim;
+}
+
+int
+get_global_offsets(const int &local_dim, std::vector<int> &offsets, const MPI_Comm &comm)
+{
+    int mpi_init;
+    MPI_Initialized(&mpi_init);
+    CAROM_VERIFY(mpi_init != 0);
+
+    int d_num_procs, d_rank;
+    MPI_Comm_rank(comm, &d_rank);
+    MPI_Comm_size(comm, &d_num_procs);
+
+    int dim = local_dim;
+    CAROM_VERIFY(MPI_Allreduce(MPI_IN_PLACE,
```

Comment:
This MPI call can be eliminated (only the second one is necessary).

Reply:
No, the first part is necessary to obtain the global size. (Note the operation is `MPI_SUM`).

Reply:
Can you just sum the entries in `offsets` to get the global size?

Reply:
I think either using `MPI_Allreduce` or summing over `offsets` is a matter of preference. Is it really necessary though?

Reply:
In general, it is good to minimize MPI calls, which have latency. In other words, small local computations are preferable to MPI calls.

Reply:
Reflected.